src/main/webapp/user_profile_script.js

```diff
@@ -83,26 +90,101 @@ async function refresh() {
     loadMessages(document.getElementById("chat-id-input").value);
 }
 
+// To address the scaling issue and avoid loading too many messages at a time,
+// the chat box would first load at most 10 most recent messages.
 async function loadMessages(keyString) {
     const queryURL = "/messages?key=" + keyString;
     const request = new Request(queryURL, {method: "GET"});
     const response = await fetch(request);
-    const msgResponse = await response.json();
+    const result = await response.json();
+    const msgResponse = result.messages;
+    cursorString = result.cursorString;
 
     const msgContainer = document.getElementById("message-container");
     msgContainer.innerHTML = "";
-    for (var index = 0; index < msgResponse.length; index++) {
-        var msg = msgResponse[index];
+    if (msgResponse.length === 10) {
+        var msg = msgResponse[9];
```

Comment:
Similar to my other review, is `var` necessary here?

Reply:
I don't think they are necessary here🤣 I would change it to `let` in the new commit.